src/Microsoft.Health.Fhir.Shared.Api/Controllers/DataConvertController.cs

```diff
@@ -0,0 +1,173 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Api.Features.Audit;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+using Microsoft.Health.Fhir.Core.Messages.DataConvert;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(AuditLoggingFilterAttribute))]
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    [ServiceFilter(typeof(ValidateContentTypeFilterAttribute))]
+    [ValidateResourceTypeFilter]
+    [ValidateModelState]
+    public class DataConvertController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly ILogger _logger;
+        private readonly DataConvertConfiguration _config;
+        private static HashSet<string> _supportedParams = GetSupportedParams();
+
+        private const char ImageRegistryDelimiter = '/';
+
+        public DataConvertController(
+            IMediator mediator,
+            IOptions<OperationsConfiguration> operationsConfig,
+            ILogger<DataConvertController> logger)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(operationsConfig?.Value?.DataConvert, nameof(operationsConfig));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _mediator = mediator;
+            _config = operationsConfig.Value.DataConvert;
+            _logger = logger;
+        }
+
+        [HttpPost]
+        [Route(KnownRoutes.DataConvert)]
+        [AuditEventType(AuditEventSubType.DataConvert)]
+        public async Task<IActionResult> DataConvert([FromBody] Parameters inputParams)
+        {
+            CheckIfDataConvertIsEnabled();
+
+            ValidateParams(inputParams);
+
+            string inputData = ReadStringParameter(inputParams, DataConvertProperties.InputData);
+            string templateCollectionReference = ReadStringParameter(inputParams, DataConvertProperties.TemplateCollectionReference);
```

Comment:
I'm not a fan of the client specifying the registry and image path. I would rather the client specify a moniker that corresponds to a template, that can be translated to an actual registry and image based on configuration.

Reply:
Me neither. The reference is not friendly for testing. The plan is to use the reference as a base format and we will enable monikers in the following updates (customers will be able to use either references or monikers). Does that make sense?

Reply:
I feel it will be better if  users can have choice to use full path or alias to retrieve templates from acr.  We are plan to add a template registering page in azure portal, which enable user add or change this mapping relationships online.  Maybe we can do it in later pr together with PaaS change.  Agree?

Reply:
As we discussed, monica support needs more investigation and will come out later after this PR. I have created a [task](https://microsofthealth.visualstudio.com/Health/_workitems/edit/77678/) to track here.